### PR TITLE
Fe 1045 eslint v9 upgrade

### PR DESCRIPTION
**Pull request by joephela**
Migrate to eslint v9 which had some major changes to how the config is handled.  Apps will need to update to the latest for all eslint related deps outlined in the readme.  They will also need to convert their .eslintrc and .eslintignore over to eslint.config.mjs.  Some of the plugins (https://github.com/typescript-eslint/typescript-eslint) we use do not yet support v9 directly so we are using a compatability layer see `FlatCompat` within eslint.config.mjs.  Once the plugins have updated we'll be able to replace the FlatCompat code with direct plugin references in the export array.

https://eslint.org/docs/latest/use/migrate-to-9.0.0
https://eslint.org/docs/latest/use/configure/migration-guide
https://acquia.atlassian.net/browse/FE-1045

### CODE REVIEW

File: packages/eslint-plugin-widen/test/jsx-import.spec.js [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
-  parserOptions: {
-    ecmaFeatures: {
-      jsx: true,
+  languageOptions: {
+    parser: tsParser,
+    parserOptions: {
+      ecmaFeatures: {
+        jsx: true,
+      },
+      ecmaVersion: 2018,
```
Comment (avo) [inline packages/eslint-plugin-widen/test/jsx-import.spec.js:13]: Do we need to specify this anymore? I believe it defaults to latest and wondering if it should be. I don't remember why we were on 2018 unless it's for backwards compatibility with something... but then we should have added a note if that were the case?

File: packages/eslint-config-widen/package.json [hunk 2]
```diff
@@ -21,14 +17,14 @@
...
+    "eslint-plugin-jest": ">= 28",
     "eslint-plugin-jsx-a11y": ">= 6",
-    "eslint-plugin-playwright": ">=0.16.0",
+    "eslint-plugin-playwright": ">=1.0.0",
     "eslint-plugin-react": ">= 7",
     "eslint-plugin-react-hooks": ">= 4",
-    "eslint-plugin-sort": ">= 2",
-    "eslint-plugin-widen": ">=2.0.0"
+    "eslint-plugin-sort": ">= 3",
+    "eslint-plugin-widen": ">=3.0.0"
```
Comment (avo) [inline packages/eslint-config-widen/package.json:27]: Can this be like th eothers?
```suggestion
    "eslint-plugin-widen": ">= 3"
```

File: packages/eslint-config-widen/package.json [hunk 3]
```diff
@@ -21,14 +17,14 @@
     "@babel/eslint-parser": "^7.22.15",
     "@typescript-eslint/eslint-plugin": "^6.9.1",
     "@typescript-eslint/parser": "^6.9.1",
-    "eslint": ">= 8",
-    "eslint-plugin-jest": ">= 27",
+    "eslint": ">= 9",
+    "eslint-plugin-jest": ">= 28",
     "eslint-plugin-jsx-a11y": ">= 6",
-    "eslint-plugin-playwright": ">=0.16.0",
+    "eslint-plugin-playwright": ">=1.0.0",
```
Comment (avo) [inline packages/eslint-config-widen/package.json:23]: Can this be
```suggestion
    "eslint-plugin-playwright": ">= 1",
```

File: packages/eslint-plugin-widen/test/jsx-import.spec.js (hunk 1 above)
Comment (joephela) [reply packages/eslint-plugin-widen/test/jsx-import.spec.js]: Removed as its probably fine on the default (latest)

File: eslint.config.mjs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import js from '@eslint/js'
+import typescriptEslint from '@typescript-eslint/eslint-plugin'
+import tsParser from '@typescript-eslint/parser'
+import globals from 'globals'
+import path from 'node:path'
+import { fileURLToPath } from 'node:url'
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+// This is in place until https://github.com/typescript-eslint/typescript-eslint
```
Comment (avo) [inline eslint.config.mjs:11]: DO we have a ticket for this so we can reference this as a TODO?

File: packages/eslint-config-widen/package.json [hunk 5]
```diff
@@ -1,15 +1,11 @@
 {
   "author": "Widen",
   "dependencies": {
-    "eslint-config-prettier": "^9.0.0"
+    "eslint-config-prettier": "^9.1.0"
   },
   "description": "Widen's shared ESLint config.",
   "exports": {
-    ".": "./lib/index.js",
-    "./jest": "./lib/jest.js",
```
Comment (avo) [inline packages/eslint-config-widen/package.json:9]: No longer needed?

File: packages/eslint-plugin-widen/package.json [hunk 6]
```diff
@@ -2,9 +2,9 @@
   "author": "Widen",
   "description": "Widen's own ESLint plugin containing custom lint rules for our code.",
   "devDependencies": {
-    "@types/eslint": "^8.44.6",
-    "@types/estree-jsx": "^1.0.2",
-    "eslint": "^8.52.0",
+    "@types/eslint": "^8.56.10",
```
Comment (avo) [inline packages/eslint-plugin-widen/package.json:5]: Are there types for eslint v9 to match or is 8.56 equivalent somehow?

File: packages/eslint-config-widen/package.json (hunk 5 above)
Comment (joephela) [reply packages/eslint-config-widen/package.json:9]: Just changed the way we export: import { base, typescript, react, playwright, jest } from 'eslint-config-widen'

File: eslint.config.mjs (hunk 4 above)
Comment (joephela) [reply eslint.config.mjs:11]: good idea, I'll add one.

File: packages/eslint-plugin-widen/package.json (hunk 6 above)
Comment (joephela) [reply packages/eslint-plugin-widen/package.json:5]: This is the latest which includes the types for the FlatConfig which was part of the eslint 9 stuff.  Must just handle versioning differently.

Comment (avo) [reply packages/eslint-plugin-widen/package.json:5]: Got it. Thanks for double checking that. It's hard to tell how different teams approach this. I've seen new major type versions that correspond with the major version and I've seen some that only update to major for real breaking changes. ¯\_(ツ)_/¯

**Pull request merged**